crates/gitbutler-diff/src/write.rs

```diff
@@ -62,7 +63,13 @@ where
             && hunks[0].diff_lines.contains_str(b"Subproject commit");
 
         // if file exists
-        if full_path.exists() {
+        if full_path.exists() || allow_new_file {
+            if hunks.len() == 1 && hunks[0].change_type == crate::ChangeType::Deleted {
```

Comment:
It feels odd to me that the single-hunk case is so special. I am sure there is a good reason for it, and I'd hope it could be explained as a comment in code. Ideally of course, there would be a test for it (and I am saying this knowing that eventually I will have to touch all of this).

(see also my comment below, I think I get it now)